src/features/Settings/components/AddOrEditNetworkNode.tsx

```diff
@@ -200,15 +212,18 @@ export default function AddOrEditNetworkNode({
     };
   }, []);
 
-  const onDeleteNodeClick = useCallback(
-    (e) => {
-      e.preventDefault();
+  const onDeleteNodeClick = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    e.preventDefault();
 
+    const network = getNetworkById(networkId);
+
+    if (network.isCustom && network.nodes.length === 1) {
+      deleteNetwork(networkId);
+    } else {
       deleteNode(editNode!.name, networkId);
-      onComplete();
-    },
-    [deleteNode, onComplete]
-  );
+    }
+    onComplete();
+  };
```

Comment:
Now that we have redux, let's dispatch an action with network id and let the reducer handle the rest.